packages/meditrak-server/src/routes/utilities/translateExpression.js

```diff
@@ -5,24 +5,27 @@
 
 /**
  * Replace question code with question UUID in an expression.
- * Also prefix UUIDs with '$' so that it can be used in mathjs later on.
+ *
+ * Important: This function also assumes that all the variables in an expression have prefix '$'
+ * because some question codes can start with numbers.
  *
  * Eg:
- * (SchFF00Section1Total / SchFF00Section1TotalObtainable) * 100
+ * ($SchFF00Section1Total / $SchFF00Section1TotalObtainable) * 100
  * will be converted to:
  * ($5ed7558b61f76a6ba5003114 / $5ec5bf8961f76a18ad00d4f9) * 100
  * @param {*} models
- * @param {*} formula
- * @param {*} codes
+ * @param {*} expression Expression to evaluate (assumes that all the variables in an expression have prefix '$')
+ * @param {*} codes: List of question codes (with no prefix $)
  */
-export const translateExpression = async (models, formula, codes) => {
-  let translatedFormula = formula;
+export const translateExpression = async (models, rawExpression, codes) => {
+  const questionCodeToId = await models.question.findCodeToId(codes);
+  let expression = rawExpression;
 
   for (const code of codes) {
-    const { id: questionId } = await models.question.findOne({ code });
+    const questionId = questionCodeToId[code];
     // UUID often starts with a number which is invalid when evaluating using mathjs
```

Comment:
```suggestion
    // Retain the $ prefix, as UUID often starts with a number (breaks evaluation in mathjs)
```
Now that the code is also prefixed with a `$`, the comment could be misleading